packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -75,8 +76,22 @@ const EditWithGeneratedProps = ( props ) => {
 	const generatedClassName = hasBlockSupport( blockType, 'className', true )
 		? getBlockDefaultClassName( name )
 		: null;
+
+	const activeVariation = getBlockTypeActiveVariation(
+		blockType.variations,
+		blockType,
+		attributes
+	);
+
+	// Generate a class name for the block's editable form.
+	const generatedVariationClassName =
+		hasBlockSupport( blockType, 'className', true ) && activeVariation
+			? getBlockDefaultClassName( `${ name }/${ activeVariation.name }` )
+			: null;
+
 	const className = clsx(
 		generatedClassName,
+		generatedVariationClassName,
```

Comment:
In my mind this should be part of `getBlockDefaultClassName`. Is there a reason for not adding the logic in that function?

Reply:
I think my initial approach did involve doing this, but IIRC @ockham preferred to keep them separated which I can also see an argument for. Not sure if you'd prefer to elaborate on this @ockham :)

Reply:
I'm thinking `getBlockDefaultClassName` is used in a number of places where the expectation is that it only returns _one_ class name. For example, the [documentation of the corresponding filter](https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#blocks-getblockdefaultclassname) states

> Generated HTML classes for blocks follow the `wp-block-{name}` nomenclature. This filter allows to provide an alternative class name.

Now it might be possible to change that, but I felt like there's too much potential for breakage of existing code. Instead, I thought it might make sense to keep `getBlockDefaultClassName` as the low-level util that generates the default `wp-block-` classname that's used by higher-level functions such as `addGeneratedClassName` to compose the final list of class names, and add `generatedVariationClassName` as another low-level util for use by the latter.

Reply:
Yeah,  I get what you mean, but still in my mind we are adding an extra default classname for every block. By having them separated it means we'd have to duplicate the checks for it, new similar filter etc.. I'd love some more thoughts maybe from @mcsf or @youknowriad ?

Reply:
I think it's fine to **not** fit everything into `getBlockDefaultClassName` (but ultimately I don't care much). 

However, here's my nit: I would consolidate the support check. It's not just saving a function call, but more importantly I think it reads better:

```js
// Generate class names for the block's editable form.
const generatedNames = [];
if ( hasBlockSupport( blockType, 'className', true ) ) {
	generatedNames.push( getBlockDefaultClassName( name ) );
	if ( activeVariation ) {
		generatedNames.push(
			getBlockDefaultClassName(
				`${ name }/${ activeVariation.name }`
			)
		);
	}
}

const className = clsx(
	generatedNames,
	attributes.className,
	props.className
);
```